ddl/db_integration_test.go

```diff
@@ -1464,19 +1464,11 @@ func (s *testIntegrationSuite3) TestAlterColumn(c *C) {
 	c.Assert(err, NotNil)
 
 	s.tk.MustExec("drop table if exists t")
-	// TODO: fix me, below sql should execute successfully. Currently, the result of calculate key length is wrong.
-	//s.tk.MustExec("create table t1 (a varchar(10),b varchar(100),c tinyint,d varchar(3071),index(a),index(a,b),index (c,d));")
-	s.tk.MustExec("create table t1 (a varchar(10),b varchar(100),c tinyint,d varchar(3068),index(a),index(a,b),index (c,d));")
-	_, err = s.tk.Exec("alter table t1 modify column a varchar(3000);")
-	c.Assert(err, NotNil)
-	c.Assert(err.Error(), Equals, "[ddl:1071]Specified key was too long; max key length is 3072 bytes")
+	s.tk.MustExec("create table t1 (a varchar(10),b varchar(100),c tinyint,d varchar(3071),index(a),index(a,b),index (c,d)) charset = ascii;")
```

Comment:
Why set charset = ascii here?

Reply:
In `ascii`, each character takes only one byte. I think this is what the author means.